### ISSUE TITLE
[Feature]: Don't require esc to create new nested tag

### ISSUE THREAD

**Issue by RangoonRen** [mgmeyers/obsidian-kanban#388]
### Goal or desired outcome of this feature

I would like to use a Kanban board to keep track of video games I want to play. I use the lists to keep track of games in progress/unplayed/need to decide if I want to play/etc. I would like to use tags to be able to say what platform a game is on. So for example one game may have the tag `#video-games/PC/GamePass`, while another might have `#video-games/PC/Steam`. The idea being that if I want to play a game on PC I can do a search on the board for `#video-games/PC` and get all my PC games. 

### Describe the feature

Currently if you try to add a nested tag to a card it doesn't allow it. As explained in the use case, I'd like to be able to add them so it can later be searched in a nested fashion.

**Comment by mgmeyers** [comment]
Thanks @RangoonRen. Could you clarify what you mean by this:

> Currently if you try to add a nested tag to a card it doesn't allow it

In what way does it not allow it? 

Nested tags seem to work ok for me, but maybe I'm missing something?

<img width="332" alt="Screen Shot 2021-11-23 at 10 17 39 AM" src="https://user-images.githubusercontent.com/2694747/143081548-cbe3ce7d-9901-4eb1-99b4-8aa06ec6a1a2.png">

**Comment by RangoonRen** [comment]
Oh I see what's happening. This should be a bug ticket then. It seems to only be an issue if the nested tag does not already exist. If I put the nested tag I want to use elsewhere first, it works just fine.

When I type the tag I get this:
![image](https://user-images.githubusercontent.com/10180022/143082247-0dd1b7a2-21d8-48b9-91f7-fde51db80539.png)

and then from there if I hit enter I get this:
![image](https://user-images.githubusercontent.com/10180022/143082345-72273c06-b43e-4075-a525-8817ddc5371f.png)



So I had already typed all this up when I realized that I can simply hit escape at the point of the first image, followed by enter to create the new ticket. I thought I had tried that before but I guess not. Sorry for the run-around
